pkg/build/builder/util.go

```diff
@@ -26,3 +30,67 @@ func getBuildEnvVars(build *api.Build) map[string]string {
 	}
 	return envVars
 }
+
+type BuildConfigClientInterface interface {
+	GetBuildConfig(namespace, name string) (*buildapi.BuildConfig, error)
+	UpdateBuildConfig(buildConfig *buildapi.BuildConfig) error
+}
+
+type BuildClientInterface interface {
```

Comment:
having seen what that has wrought, i really disagree with that convention.  it has created a nightmare of interfaces that make it really hard to follow what is being implemented, not to mention redundant logic all over the place.

I also don't really see that this reduces local testability, as you can see in the PR i've retained all the existing tests.